web/api/backend.go

```diff
@@ -113,45 +114,58 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		// If the structure of the body is wrong, return an HTTP error.
 		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "Unable to Parse JSON")
 		return
 	}
 
-	// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
-	if (requestBody.UseToken && len(requestBody.Token) == 0) || (len(requestBody.Credential) == 0) {
-		w.WriteHeader(http.StatusBadRequest)
+	// JSON BODY representing the credentials attribute to be used (i.e. either the credentials found in the token or inputted).
+	creds := make(map[string]interface{})
+
+	// JSON Body representing the credentials attribute with the following format: "credentials":{"scopes":[],...}
+	// This is so to match with how the wrapper class will save the credentials file to memory.
+	credsString := make(map[string]interface{})
+	credsString = nil
```

Comment:
Not sure why this gets overwritten here, any reason for that?

Reply:
golang apparently does not like it if a variable is initialized with nil. I overwrote with nil so that if the command type is test, info, or reset, the credentials string will be nil (as per wrapper design), and I don't have to do another logic statement to change it at the end. Otherwise, the credentials string will be changed in the if statement.